packages/components/src/components/Button/Button.tsx

```diff
@@ -23,29 +25,72 @@ export const Button = React.forwardRef<View, ButtonProps & ViewProps>(
         },
         ref
     ) => {
+        const toggleData = useContext(ToggleButtonContext);
+        let isToggleButton = false;
+        if (toggleData && toggleData.valid) {
+            isToggleButton = true;
+            variant = toggleData.isSelected ? "contained" : "outlined";
+        }
         const styles = useStyles(themeStyles, { color, variant });
+        const style: any[] = [styles.colorContainer];
+
+        const groupData = useContext(ButtonGroupContext);
+        if (groupData && groupData.valid) {
+            if (groupData.index === 0) {
+                style.push(buttonGroupStyles.first);
+            } else if (groupData.index === groupData.length - 1) {
+                style.push(buttonGroupStyles.last);
+            } else {
+                style.push(buttonGroupStyles.middle);
+            }
+        }
+        style.push(rest.style);
 
         return (
-            <View ref={ref} style={[styles.colorContainer, rest.style]}>
-                <PressableHighlight
-                    onPress={onPress}
-                    style={styles.pressableContainer}
-                >
+            <View ref={ref} style={style}>
+                {isToggleButton ? <View style={styles.pressableContainer}>
```

Comment:
I see that you are defining a `Pressable` wrapper in the ToggleButton and omitting the `onPress` callback here. Would it be possible to expose a callback in the context named `onToggle()` or something, instead?
This way, you don't need two different views being rendered, only a conditional callback invokation: `onPress={toggleData ? toggleData.onToggle : onPress}`